client/homebrew/brewRenderer/brewRenderer.jsx

```diff
@@ -76,9 +75,80 @@ const BrewRenderer = (props)=>{
 		rawPages = props.text.split(/^\\page$/gm);
 	}
 
-	useEffect(()=>{ // Unmounting steps
-		return ()=>{window.removeEventListener('resize', updateSize);};
+	useEffect(() => {
+		const locationHash = window.location.hash;
+		const iframe = document.getElementById('BrewRenderer');
+		
+		const pageIdRegex = /^#p\d+$/;
+		
+		const handleIframeLoad = () => {
+			setTimeout(() => {
+				if (pageIdRegex.test(locationHash)) {
+					const pageNumber = parseInt(locationHash.slice(2));
+					scrollToPage(pageNumber - 1);
+				} else {
+					const elementId = locationHash.slice(1);
+					if (elementId) {
+						getPageContainingElement(elementId)
+							.then((pageNumber) => {
+								if (pageNumber !== -1) {
+									scrollToPage(pageNumber);
```

Comment:
Do we not want to just scroll to the element directly instead of the containing page? Would simplify logic.

Reply:
Myself i prefer scrolling to the container page, much more clean experience. But i am not 100% settled on that

Reply:
I have found scrolling to the page is REALLY annoying on big documents.

Reply:
my experience is different, but this change is not worth fighting over, will do this change